### ISSUE TITLE
NavLink can't call context.makePath()

### ISSUE THREAD

**Issue by koichik** [yahoo/fluxible#144]
After #142 was merged, `NavLink` can't call `context.makePath()`.

```
$ yo fluxible
...
$ npm i fluxible@0.4.5
...
$ grunt
...
/tmp/example/node_modules/flux-router-component/lib/NavLink.js:52
            href = context.makePath(routeName, props.navParams);
                           ^
TypeError: undefined is not a function
    at [object Object].React.createClass._getHrefFromProps (/tmp/example/node_modules/flux-router-component/lib/NavLink.js:52:28)
    at [object Object].React.createClass.getInitialState (/tmp/example/node_modules/flux-router-component/lib/NavLink.js:39:24)
    at new Constructor (/tmp/example/node_modules/react/lib/ReactClass.js:853:54)
    at [object Object].ReactCompositeComponentMixin.mountComponent (/tmp/example/node_modules/react/lib/ReactCompositeComponent.js:135:16)
    at [object Object].wrapper [as mountComponent] (/tmp/example/node_modules/react/lib/ReactPerf.js:70:21)
    at Object.ReactReconciler.mountComponent (/tmp/example/node_modules/react/lib/ReactReconciler.js:38:35)
    at ReactDOMComponent.ReactMultiChild.Mixin.mountChildren (/tmp/example/node_modules/react/lib/ReactMultiChild.js:192:44)
    at ReactDOMComponent.Mixin._createContentMarkup (/tmp/example/node_modules/react/lib/ReactDOMComponent.js:287:32)
    at ReactDOMComponent.Mixin.mountComponent (/tmp/example/node_modules/react/lib/ReactDOMComponent.js:197:12)
    at Object.ReactReconciler.mountComponent (/tmp/example/node_modules/react/lib/ReactReconciler.js:38:35)
```

It seems `ComponentContext` isn't propagated to child components.

**Comment by tbo** [comment]
I can confirm this. Since 0.4.4 the component context doesn't contain makePath anymore.

**Comment by ali1k** [comment]
I think the `RouterMixin` needs to be enabled on the component.

**Comment by mridgway** [comment]
Ok, so the `FluxibleComponent` does need to pass `context` down to the child component so that the `RouterMixin` can get access to `this.props.context`.

Thanks for reporting guys. Fix incoming.

**Comment by mridgway** [comment]
This should be fixed in v0.4.6.

**Comment by koichik** [comment]
Confirmed, thanks!
